src/mobdebug.lua

```diff
@@ -511,8 +511,8 @@ local function handle_breakpoint(peer)
   if buf:sub(2,2) ~= 'E' then return end
 
   -- need to read few more characters
-  buf = buf .. readnext(peer, 5-#buf)
-  if buf ~= 'SETB ' and buf ~= 'DELB ' then return end
+  if #buf < 5 then buf = buf .. readnext(peer, 5-#buf) end
+  if buf:sub(1,5) ~= 'SETB ' and buf:sub(1,5) ~= 'DELB ' then return end
 
   local res, _, partial = peer:receive("*l") -- get the rest of the line; blocking
```

Comment:
Just in case. Is it possible that the `buf` already has full line?

Reply:
I don't think so, as so far we only read at most 2 bytes (or had some leftovers in the buffer that are at least 2 bytes long). We also know that the content of the buffer starts with `SE` or `DE` and there are no full commands that could match that content.

Reply:
I don't think it's possible that `buf` already has a full line, but it's possible to have longer than 5 bytes string in `buf`, which may cause a run-time error on line 514.